packages/core/src/awsService/appBuilder/wizards/serverlessLandWizard.ts

```diff
@@ -0,0 +1,290 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as AWS from '@aws-sdk/types'
+import * as vscode from 'vscode'
+import { Wizard } from '../../../shared/wizards/wizard'
+import * as path from 'path'
+import { createInputBox } from '../../../shared/ui/inputPrompter'
+import { createCommonButtons } from '../../../shared/ui/buttons'
+import { createQuickPick } from '../../../shared/ui/pickerPrompter'
+import { createFolderPrompt } from '../../../shared/ui/common/location'
+import { Region } from '../../../shared/regions/endpoints'
+import { createExitPrompter } from '../../../shared/ui/common/exitPrompter'
+import * as nodefs from 'fs' // eslint-disable-line no-restricted-imports
+
+export interface CreateServerlessLandWizardForm {
+    name: string
+    location: vscode.Uri
+    pattern: string
+    runtime: string
+    iac: string
+    region?: string
+    schemaName?: string
+}
+interface IaC {
+    id: string
+    name: string
+}
+interface Runtime {
+    id: string
+    name: string
+    version: string
+}
+interface PatternData {
+    name: string
+    description: string
+    runtimes: Runtime[]
+    iac: IaC[]
+}
+
+export interface ProjectMetadata {
+    patterns: Record<string, PatternData>
+}
+
+/**
+ * Manages metadata for serverless application patterns
+ * Implements Singleton pattern to ensure single instance across the application
+ */
+export class MetadataManager {
```

Comment:
I'm not sure where, but it might make sense for this class to live in a different file and not inside "serverlessLandWizard". Maybe in its own file inside "models" too?

Reply:
I have added this inside a separate file `serverlesslandMetadataManager.ts`. Please suggest another naming if you think this is not correct.